libcloud/compute/drivers/dimensiondata.py

```diff
@@ -638,12 +639,18 @@ def ex_power_off(self, node):
         """
         request_elm = ET.Element('powerOffServer',
                                  {'xmlns': TYPES_URN, 'id': node.id})
-        body = self.connection.request_with_orgId_api_2(
-            'server/powerOffServer',
-            method='POST',
-            data=ET.tostring(request_elm)).object
-        response_code = findtext(body, 'responseCode', TYPES_URN)
-        return response_code in ['IN_PROGRESS', 'OK']
+
+        try:
+            body = self.connection.request_with_orgId_api_2(
+                'server/powerOffServer',
+                method='POST',
+                data=ET.tostring(request_elm)).object
+            response_code = findtext(body, 'responseCode', TYPES_URN)
+        except (DimensionDataAPIException, NameError, BaseHTTPError):
+            r = self.ex_get_node_by_id(node.id)
+            response_code = r.state.upper()
+
+        return response_code in ['IN_PROGRESS', 'OK', 'STOPPED', 'STOPPING']
```

Comment:
This looks good to me, but maybe we should consider throwing an exception if node is already in stopping / stopped state.

That's more just me thinking out loud, I'm also fine with current approach.